try {} catch (error: unknown) {}
try {} catch (error) {}
              ~~~~~ [implicitAny]
try {} catch (error: any) {}
              ~~~~~~~~~~ [explicitAny]

Promise.reject("Kaboom!").catch(
  () => console.error("Whoops!")
);
Promise.reject("Kaboom!").catch(
  function () { console.error("Whoops!"); }
);

Promise.reject("Kaboom!").catch(
  (error: unknown) => console.error(error)
);
Promise.reject("Kaboom!").catch(
  function (error: unknown) { console.error(error); }
);

Promise.reject("Kaboom!").then(
  () => {},
  (error: unknown) => console.error(error)
);
Promise.reject("Kaboom!").then(
  function () {},
  function (error: unknown) { console.error(error); }
);

Promise.reject("Kaboom!").catch(
  (error) => console.error(error)
   ~~~~~ [implicitAny]
);
Promise.reject("Kaboom!").catch(
  function (error) { console.error(error); }
            ~~~~~ [implicitAny]
);

Promise.reject("Kaboom!").catch(
  (error: any) => console.error(error)
   ~~~~~~~~~~ [explicitAny]
);
Promise.reject("Kaboom!").catch(
  function (error: any) { console.error(error); }
            ~~~~~~~~~~ [explicitAny]
);

Promise.reject("Kaboom!").then(
  () => {},
  (error) => console.error(error)
   ~~~~~ [implicitAny]
);
Promise.reject("Kaboom!").then(
  function () {},
  function (error) { console.error(error); }
            ~~~~~ [implicitAny]
);

Promise.reject("Kaboom!").then(
  () => {},
  (error: any) => console.error(error)
   ~~~~~~~~~~ [explicitAny]
);
Promise.reject("Kaboom!").then(
  function () {},
  function (error: any) { console.error(error); }
            ~~~~~~~~~~ [explicitAny]
);

[explicitAny]: Explicit any
[implicitAny]: Implicit any
